packages/realm/src/List.ts

```diff
@@ -279,4 +279,38 @@ export class List<T = unknown> extends OrderedCollection<T> implements Partially
     }
     return result;
   }
+
+  remove(index: number) {
+    assert.inTransaction(this.realm);
+    assert.number(index, "index");
+    const { internal } = this;
+
+    //TODO I am not sure exactly if we expect a no-op for out of bounds operations or an exception
+    assert(index >= 0, "Index cannot be smaller than 0");
+    assert(index < internal.size, "Index cannot be bigger than the size of the list");
+
+    internal.remove(index);
+  }
+
+  move(from: number, to: number) {
+    assert.inTransaction(this.realm);
+    assert.number(from, "index");
+    assert.number(to, "to");
+
+    //TODO Need to decide if out-of-bounds indexes will be a no-op or an exception
+    const { internal } = this;
+
+    internal.move(from, to);
```

Comment:
This I would definitely not de-structure, because it makes `internal` appear twice instead of once.

```suggestion
    this.internal.move(from, to);
```

Reply:
It's also being done elsewhere outside of this PR (see `splice` above for example).

Reply:
Yeah. And I'm all for that, because it has 9 occurrences of internal it simplifies the code. In this case it doesn't serve that purpose.

Reply:
Makes sense. I've removed the destructuring.